File: tests/statistical_tests/spot_checks/t-test-one-sample.test.R
#!/usr/bin/env Rscript

cat("\n📊 One-Sample T-Test\n")
cat(paste(rep("-", 80), collapse = ""), "\n")

library(effsize)

data <- c(12.5, 13.1, 11.8, 12.9, 13.3, 12.2, 12.7, 13.0)
mu <- 12.0

# One-sample t-test
result <- t.test(data, mu = mu, alternative = "two.sided", conf.level = 0.95)

# Cohen's d for one-sample test
cohens_d <- cohen.d(data, NA, mu = mu)

cat("Data:", data, "\n")
cat("mu (null hypothesis):", mu, "\n")
cat("t-statistic:", result$statistic, "\n")
cat("df:", result$parameter, "\n")
cat("p-value:", result$p.value, "\n")
cat("CI lower:", result$conf.int[1], "\n")
cat("CI upper:", result$conf.int[2], "\n")
cat("Cohen's d:", cohens_d$estimate, "\n")

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("ONE-SAMPLE T-TEST SPOT CHECK\n")
cat(paste(rep("=", 80), collapse = ""), "\n")

File: tests/statistical_tests/spot_checks/t-test-one-sample.test.ts
import { stats } from "../../../src/dataframe/mod.ts";

Deno.test("One-Sample T-Test Spot Check", () => {
  console.log("\n📊 One-Sample T-Test");
  console.log("-".repeat(80));

  const data = [12.5, 13.1, 11.8, 12.9, 13.3, 12.2, 12.7, 13.0];
  const mu = 12.0;

  const result = stats.test.t.oneSample({
    data,
    mu,
    alternative: "two-sided",
  });

  console.log("Data:", data);
  console.log("mu (null hypothesis):", mu);
  console.log("t-statistic:", result.test_statistic.value);
  console.log("df:", result.degrees_of_freedom);
  console.log("p-value:", result.p_value);
  console.log("CI:", [
    result.confidence_interval.lower,
    result.confidence_interval.upper,
  ]);
  console.log("Cohen's d:", result.effect_size.value);
});

console.log("\n" + "=".repeat(80));
console.log("ONE-SAMPLE T-TEST SPOT CHECK");
console.log("=".repeat(80));
